route/grafananet.go

```diff
@@ -234,16 +236,37 @@ func (route *GrafanaNet) retryFlush(metrics []*schema.MetricData, buffer *bytes.
 	return metrics[:0]
 }
 
-func (route *GrafanaNet) flush(req *http.Request) (time.Duration, error) {
+func (route *GrafanaNet) flush(mda schema.MetricDataArray, req *http.Request) (time.Duration, error) {
 	pre := time.Now()
 	resp, err := route.client.Do(req)
 	dur := time.Since(pre)
 	if err != nil {
 		return dur, err
 	}
 	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
-		ioutil.ReadAll(resp.Body)
+		bod, err := ioutil.ReadAll(resp.Body)
 		resp.Body.Close()
+		if err != nil {
+			log.Warnf("GrafanaNet processed the request, but could not read the response: %s", err.Error())
+			return dur, nil
+		}
+		var resp MetricsResponse
+		err = json.Unmarshal(bod, &resp)
+		if err != nil {
+			log.Warnf("GrafanaNet processed the request, but could not parse the response: %s", err.Error())
```

Comment:
Here and on line 250 the messages read as though the server couldn't read/parse the response.  Maybe something like `Received status code <resp.StatusCode> from server, but response body could not be <read/parsed>: <error>` would be clearer.